source/models/bounties/Bounty.js

```diff
@@ -0,0 +1,124 @@
+﻿const { EmbedBuilder, Guild } = require('discord.js');
+const { DataTypes: { BIGINT, STRING, INTEGER, BOOLEAN }, Model } = require('sequelize');
+const { ihpAuthorPayload } = require('../../embedHelpers');
+const { Hunter } = require('../users/Hunter');
+const { Guild: HunterGuild } = require('../guilds/Guild');
+const { database } = require('../../../database');
+
+/** Bounties are user created objectives for other server members to complete */
+const bountyModel = {
+	id: {
+		primaryKey: true,
+		type: BIGINT,
+		autoIncrement: true
+	},
+	userId: {
+		type: STRING,
+		references: {
+			model: 'User',
+			key: 'id'
+		}
+	},
+	guildId: {
+		type: STRING,
+		references: {
+			model: 'Guild',
+			key: 'id'
+		}
+	},
+	postingId: {
+		type: STRING,
+	},
+	slotNumber: {
+		type: INTEGER,
+		allowNull: false
+	},
+	isEvergreen: {
+		type: BOOLEAN,
+		devaultValue: false
+	},
+	title: {
+		type: STRING,
+		allowNull: false,
+		defaultValue: "\u200B"
+	},
+	description: {
+		type: STRING,
+		allowNull: false,
+		defaultValue: "\u200B"
+	},
+	attachmentURL: {
+		type: STRING,
+		defaultValue: null
+	},
+	scheduledEventId: {
+		type: INTEGER,
+		defaultValue: null
+	},
+	state: { // Allowed values: "open", "completed", "deleted"
+		type: STRING,
+		defaultValue: "open"
+	},
+	completedAt: {
+		type: INTEGER,
+		defaultValue: null
+	},
+	deletedAt: {
+		type: INTEGER,
+		defaultValue: null
+	},
+	editCount: {
+		type: INTEGER,
+		defaultValue: 0
+	}
+};
+exports.Bounty = class Bounty extends Model {
+	/** Generate an embed for the given bounty
+	 * @param {Guild} guild
+	 * @param {Hunter?} hunter
+	 * @param {HunterGuild?} hunterGuild
+	 */
+	asEmbed(guild, hunter, hunterGuild) {
+		return guild.members.fetch(this.userId).then(async author => {
+			if (!hunter) {
+				hunter = await database.models.Hunter.findOne({ where: { userId: this.userId, guildId: this.guildId } });
+			}
+			if (!hunterGuild) {
+				hunterGuild = await database.models.Guild.findByPk(this.guildId);
```

Comment:
Since the Bounty doesn't really "own" the hunter or guild, I'd probably just prefer if we always pass the hunter and hunterGuild parameters and throw an error in these spots, instead.

Reply:
I consider this temporary (maybe I should have commented as so?), with a future investigation being "can we always pass the hunter and hunterGuild?" to be done when everything is implemented. In the "can always pass" world, I would probably even skip out on error throwing.